Sort optional columns in documentation (#292)
EducationalTestingService/rsmtool: doc/config_rsmtool.rst

"""""""""
The path to the evaluation data feature file in one of the :ref:`supported formats <input_file_format>`. Each row should correspond to a single response and contain numeric feature values extracted for this response. In addition, there should be a column with a unique identifier (ID) for each response and a column with the human score for each response. The path can be absolute or relative to the location of config file.

.. note ::

>     1. By default, ``rsmtool`` will use all of the columns present in the training and evaluation files as features except for any columns optionally identified as containing other information (e.g., :ref:`flag_column <flag_column_rsmtool>`, :ref:`id_column <id_column_rsmtool>`, :ref:`train_label_column <train_label_column_rsmtool>` et cetera below). 

Reviewer: ```suggestion
    1. By default, `rsmtool` will use `spkitemid` column as the unique id for each row, `sc1` column as the training label and all other columns present in the training and evaluation files as features. Optional fields :ref:`id_column <id_column_rsmtool>` and  :ref:`train_label_column <train_label_column_rsmtool>` can be used to specify different names for the id column and label column. In addition, configuration file can also be used to identify columns containing other information (e.g., :ref:`second_human_score_column <second_human_score_column_rsmtool>`,  :ref:`flag_column <flag_column_rsmtool>`, :ref:`subgroups <subgroups_rsmtool>` et cetera below). Any remaining column that is not defined in the configuration file will be considered a feature column. See below if you only want to use a subset of remaining column.
```

"""""""""
The path to the evaluation data feature file in one of the :ref:`supported formats <input_file_format>`. Each row should correspond to a single response and contain numeric feature values extracted for this response. In addition, there should be a column with a unique identifier (ID) for each response and a column with the human score for each response. The path can be absolute or relative to the location of config file.

.. note ::

>     1. For both the training and evaluation files, the default behavior of ``rsmtool`` is to look for columns named ``spkitemid`` and ``sc1`` in order to get the unique IDs and the train/test labels for each response, respectively. The optional fields :ref:`id_column <id_column_rsmtool>` and  :ref:`train_label_column <train_label_column_rsmtool>` can be used to specify different names for the id and label columns respectively. 

Reviewer: ```suggestion
    1. For both the training and evaluation files, the default behavior of ``rsmtool`` is to look for a column named ``spkitemid`` in order to get the unique IDs for each response and column named ``sc1`` to get train/test labels. The optional fields :ref:`id_column <id_column_rsmtool>` and  :ref:`train_label_column <train_label_column_rsmtool>` can be used to specify different names for these columns. 
```

>     2. ``rsmtool`` also assumes that any columns present in these files (other than those containing IDs and labels) contain feature values. If this is not the case, one can identify columns containing other, non-feature information, e.g., :ref:`second_human_score_column <second_human_score_column_rsmtool>`, :ref:`flag_column <flag_column_rsmtool>`, :ref:`subgroups <subgroups_rsmtool>` et cetera below. 

Reviewer: ```suggestion
    2. ``rsmtool`` also assumes that all other columns present in these files (other than those containing IDs and labels) contain feature values. If this is not the case, one can use the configuration file to identify columns containing other, non-feature information that will be used for the analyses, e.g., :ref:`second_human_score_column <second_human_score_column_rsmtool>`, :ref:`flag_column <flag_column_rsmtool>`, :ref:`subgroups <subgroups_rsmtool>` et cetera below. 
```

""""""""""""""""""""""""
A brief description of the experiment. This will be included in the report. The description can contain spaces and punctuation. It's blank by default.

.. _exclude_zero_scores_rsmtool:

exclude_zero_scores *(Optional)*
""""""""""""""""""""""""""""""""
By default, responses with human scores of 0 will be excluded from both training and evaluation set. Set this field to ``false`` if you want to keep responses with scores of 0. Defaults to ``true``.

.. _feature_subset:

feature_subset *(Optional)*
"""""""""""""""""""""""""""
Name of the pre-defined feature subset to be used if using :ref:`subset-based column selection <subset_column_selection>`.

.. _feature_subset_file:

feature_subset_file *(Optional)*
""""""""""""""""""""""""""""""""
Path to the feature subset file if using :ref:`subset-based column selection <subset_column_selection>`.

.. _feature_file_rsmtool:

features *(Optional)*
"""""""""""""""""""""
Path to the file with list of features if using :ref:`fine-grained column selection <feature_list_column_selection>`. Alternatively, you can pass a ``list`` of feature names to include in the experiment.

.. _feature_fields_note:

.. note ::

>     By default, ``rsmtool`` will use all of the columns present in the training and evaluation files as features except for any columns explicitly identified in the configuration file (see below). The following four fields (``features``, ``feature_subset_file``, ``feature_subset``, and ``sign``) are useful if you want to use only a specific set of columns as features. See :ref:`selecting feature columns <column_selection_rsmtool>` for more details.

Author: I agree that the note should be moved. I am going to move it to right under `test_file` since the note refers to "training and evaluation files". I am also going to break up the note into two points, make the literals into links instead, and tweak the language a bit.
